src/routers/alpha-router/alpha-router.ts

```diff
@@ -1955,6 +1988,12 @@ export class AlphaRouter
 
     const quotePromises: Promise<GetQuotesResult>[] = [];
 
+    // for v4, for now we explicitly require people to specify
+    if (v4ProtocolSpecified) {
```

Comment:
shouldnt we do this before the rpoise? otherwise we are just wasting resources right?

Reply:
I think I didn't follow. Maybe we can chat more offline.

Reply:
My understanding is that the upstream dependencies will always specify a protocol version when they request for a route. I think this if block shouldn't get entered, since v4 is not supported yet. Not sure if answers?